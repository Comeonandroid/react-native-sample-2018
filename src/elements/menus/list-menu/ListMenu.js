import React from 'react'
import styled from 'styled-components'
import { PropTypes } from 'prop-types'

import { View } from '../../index'

const ListMenu = ({ children, ...rest }) => (
  <Element {...rest}>
    {children}
  </Element>
)

ListMenu.propTypes = {
  children: PropTypes.node
}

const Element = styled(View) ``
         
export default ListMenu