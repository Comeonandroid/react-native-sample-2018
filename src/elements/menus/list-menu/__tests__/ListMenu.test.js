import 'react-native'
import React from 'react'
import { itBehavesLike } from '../../../../specs/shared_examples/index'

import ListMenu from '../ListMenu'

describe('<ListMenu />', () => {
  itBehavesLike('aThemedComponent', { subject: <ListMenu /> })
  itBehavesLike('aComponentWithChildren', { subject: <ListMenu /> })
})
