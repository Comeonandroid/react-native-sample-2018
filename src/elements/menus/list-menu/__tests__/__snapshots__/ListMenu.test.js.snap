// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListMenu /> dark: renders correctly 1`] = `
<View
  style={
    Array [
      Object {},
      Array [
        Object {},
        undefined,
      ],
    ]
  }
/>
`;

exports[`<ListMenu /> light: renders correctly 1`] = `
<View
  style={
    Array [
      Object {},
      Array [
        Object {},
        undefined,
      ],
    ]
  }
/>
`;

exports[`<ListMenu /> renders component when passed in 1`] = `
<Styled(View)>
  <Component />
</Styled(View)>
`;

exports[`<ListMenu /> renders text when passed in 1`] = `
<Styled(View)>
  Test
</Styled(View)>
`;
